fix: allow caching pages on different desk partitions

alethyst opened the pull request:
Hello @eneajaho 

I ran into an issue using file system caching when deploying on AWS Lambda, file system write access is blocked on lambda so it's not possible to store the cache in the build directory, but they do provide a `/tmp` partition to store any thing on it. So I changed the caching directory path like this:

```typescript
const isRunningOnLambda = !!(process.env.LAMBDA_TASK_ROOT || process.env.AWS_EXECUTION_ENV);
const isrCacheFolderPath = isRunningOnLambda ? join('/tmp', '/isr-cache') : join(distFolder, '/isr-cache');
```
<br>

But now I got a new error message which is 
```
EXDEV: cross-device link not permitted, rename '/var/task/dist/browser/index.html' -> '/tmp/isr-cache/__.html'`
```
<br>

A bit cryptic, but it's coming from [filesystem-cache-handler.ts](https://github.com/eneajaho/ngx-isr/blob/v.0.5.4/projects/ngx-isr/src/lib/cache-handlers/filesystem-cache-handler.ts) at line 248
```typescript
fs.renameSync(path, newFilePath);
```
And after a bit of research it turns out `rename` doesn't allow renaming files across partitions so it fails.
<br>

The fix turns out to be fairly simple, all we have to is replace line 248 in [filesystem-cache-handler.ts](https://github.com/eneajaho/ngx-isr/blob/v.0.5.4/projects/ngx-isr/src/lib/cache-handlers/filesystem-cache-handler.ts) with: 
```typescript
fs.copyFileSync(path, newFilePath);
fs.rmSync(path);
```
<br>

I've already implemented a custom handler for my case but it would be awesome if I didn't have to do that, I also don't think consumers of the package should have to worry about low level details like this, so I've created this PR for it.

eneajaho commented:
Hello @alethyst 
Thanks for creating this PR 🎉! Appreciate it! 

Will release a new version of package later today and will let you know!

alethyst commented:
Awesome, thank you for merging 🙌

eneajaho commented:
Hello @alethyst 
Your PR was released in the new version (https://github.com/eneajaho/ngx-isr/releases/tag/v.0.5.5) 
But it includes some small breaking changes [imports changes] that should be easy to change. 

Thanks again for the contribution 🙌! 

Let me know for any other issue you may encounter.

alethyst commented:
Hey @eneajaho 

Thank you for the quick release 🎉

I've updated to the latest version and updated the import paths and everything is working great!

One thing I've noticed though is that the release notes mentions that it's a breaking change for the import path, but when I updated the package, the imports that I already had also worked just fine since these exports are still there:

![image](https://user-images.githubusercontent.com/39350029/236346133-1749aeb4-1a5d-48a7-8b5f-d336f1d9d94b.png)

Not sure if this is intended but I just though I'd mention it.
